MathLib/LinAlg/PETSc/PETScVector.h

```diff
@@ -171,6 +177,14 @@ class PETScVector
             VecGetValues(_v, e_idxs.size(), &e_idxs[0], &sub_vec[0]);
         }
 
+        // TODO preliminary
+        double operator[] (PetscInt idx) const
```

Comment:
I know. The problem is that the interfaces how to access values in EigenVector, Eigen::VectorXd, and PETScVector all are different. I need the access in `Tests/NumLib/ODEs.h` and I did not want to add another method like `setMatrix()` to `ODETypes.h`. It seemed to be the minimal change to just add `operator[]` to the PETScVector. That's not a final decision, that's why it's marked preliminary.
I don't want to clutter this PR with more of this interface unification stuff. Once we design a uniform matrix and vector interface (hopefully soon), we will remove that method. But for the time being I'd like to keep it in this PR.